ethcore/engines/validator-set/src/safe_contract.rs

```diff
@@ -16,9 +16,10 @@
 
 /// Validator set maintained in a contract, updated using `getValidators` method.
 
-use std::sync::{Weak, Arc};
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex, Weak};
```

Comment:
why not `parking_lot::Mutex`? we're using `parking_lot::RwLock` here already, this would allow us to get rid of `unwrap`s